design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,199 @@
+# Ignore changes
+
+* Owners: Lovro Sviben (@lsviben)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+When Crossplane manages a resource, it reconciles all of the parameters under `spec.forProvider` to the external providers both 
+during the creation and subsequent updates to the managed resource. But in some cases, a parameter of the external resource
+could change due to it being managed outside of Crossplane. In the current state, if that happens, Crossplane will try to 
+"fix" the situation, and update the parameter back to the one it has in the `spec.forProvider` as it is designed to be
+the only source of truth.
+
+Some examples of this case:
+- [AutoScalingGroup DesireCapacity], where when using an autoscaler, the parameter gets changed 
+externally.
+- [DynamoDB Tables] with using autoscaling ([Policy] + [Target]). The autoscaler scales the read and write capacity,
+and then Crossplane resets it back to the `readCapacity` and `writeCapacity` defined on the Table, which effectively makes autoscaling useless.
+- [NodeGroup] while using a node autoscaler. There is a race condition between crossplane and node autoscaler. 
+Node autoscaler increases the `desiredSize` due to resource requirement but crossplane decrease it to `desiredSize` set in composition.
+
+In these cases, it would make sense to make Crossplane ignore select parameters during updates. 
+Terraform addresses this use case with its [ignore_changes] functionality. 
+
+An additional benefit of ignoring some fields would be that it would be possible to have different Crossplane
+Compositions managing a single external resource. This could open up some interesting new possibilities for Crossplane.
+
+So in this document, the aim will be to introduce a solution in Crossplane for ignoring some managed resource parameters
+during updates.
+
+## Goals
+
+* Design a functionality that allows for ignoring some fields during resource updates.
+* Try and make it as easy as possible to apply the ignoring functionality on existing Crossplane providers. In other words, to 
+design a standard way on how to update the providers code to support `ignore changes`.
+
+## Solution
+
+The idea is to introduce a new `ignoreChanges` string array field in the `spec` of the managed resource, that would work similarly
+to Terraform's [ignore_changes]. The items in the array would be the field paths of the fields that should
+be ignored on updates. 
+
+```yaml
+apiVersion: eks.aws.crossplane.io/v1alpha1
+kind: NodeGroup
+metadata:
+  name: my-group
+spec:
+  ignoreChanges:
+    - scalingConfig.desiredSize
```

Comment:
Would this have any interaction with the new `managementPolicy` field?

It seems like a bit of a shame to introduce two new concepts (`managementPolicy` and `ignoreChanges`) to address what to me seem like two very similar / closely related problems (essentially OO resources and OO fields).

Reply:
The interaction would be that if the `managementPolicy` : `ObserveOnly` is set, then there is no point to set `ignoreChanges`. 

You are right, these problems are similar, as you say OO resources and OO fields. `ignoreChanges` just gives a more fine-grained control, and tries to fix a bit different problem, where ownership of an external resource is shared. While ObserveOnly is either owned by Crossplane or by something else. 

Not sure currently how we could bring this concepts closer together, to avoid having different solutions for similar problems.